### PR TITLE
9.0 mail cleanup

### PR DESCRIPTION
**Pull request by mdietrichc2c**
Migration to 9.0 of the module `mail_cleanup` , which existed only in 7.0 : https://github.com/OCA/server-tools/tree/7.0/mail_cleanup

### CODE REVIEW

File: mail_cleanup/README.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+* None
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/server-tools/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/server-tools/issues/new?body=module:%20mail_cleanup%0Aversion:%207.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (yvaucher) [inline mail_cleanup/README.rst]: s/7.0/9.0

File: mail_cleanup/README.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (yvaucher) [inline mail_cleanup/README.rst]: s/http/https

File: mail_cleanup/README.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+Mail cleanup
+===========
```
Comment (yvaucher) [inline mail_cleanup/README.rst:7]: Missing a = and main title is now:

```
============
Mail cleanup
============
```

File: mail_cleanup/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Mail cleanup',
+    'version': '9.0.1.0.0',
+    'category': 'Tools',
+    'summary': 'Mark as read or delete mails after a set time',
+    'author': "Camptocamp,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'website': 'http://odoo-community.org',
```
Comment (yvaucher) [inline mail_cleanup/__openerp__.py]: s/http/https

**Comment by yvaucher** [comment]
Only some nitpicking in the README.rst otherwise :+1:

**Comment by mdietrichc2c** [comment]
Thanks for the review ; I corrected those.

**Comment by yvaucher** [comment]
Travis is not happy:

```
File "/home/travis/build/OCA/server-tools/mail_cleanup/__init__.py", line 1, in <module>
from . import models
ImportError: cannot import name models
```

**Comment by mdietrichc2c** [comment]
@yvaucher sorry, missing a file ; it's now fixed.

File: mail_cleanup/models/mail_cleanup.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            # convert string values to integer
+            if config_vals['cleanup_days']:
+                config_vals['cleanup_days'] = int(config_vals['cleanup_days'])
+            if config_vals['purge_days']:
+                config_vals['purge_days'] = int(config_vals['purge_days'])
+
+            for field in ['cleanup_days', 'purge_days', 'cleanup_folder']:
+                fetchmail[field] = config_vals[field]
+
+    cleanup_days = fields.Integer(
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: Field definitions should be above methods

File: mail_cleanup/models/mail_cleanup.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        compute=_get_cleanup_conf,
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: Please use string for computation declarations in order to allow inherited changes: `compute='_get_cleanup_conf',`

File: mail_cleanup/models/mail_cleanup.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        string='Expiration days',
+        help="Number of days before marking an e-mail as read")
+
+    cleanup_folder = fields.Char(
+        compute=_get_cleanup_conf,
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: Please use string for computation declarations in order to allow inherited changes: `compute='_get_cleanup_conf',`

File: mail_cleanup/models/mail_cleanup.py [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        string='Expiration folder',
+        help="Folder where an e-mail marked as read will be moved.")
+
+    purge_days = fields.Integer(
+        compute=_get_cleanup_conf,
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: Please use string for computation declarations in order to allow inherited changes: `compute='_get_cleanup_conf',`

File: mail_cleanup/models/mail_cleanup.py [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        string='Deletion days',
+        help="Number of days before removing an e-mail")
+
+    def _cleanup_fetchmail_server(self, server, imap_server):
+        count, failed = 0, 0
+        expiration_date = fields.Date.from_string(fields.Date.today())
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: `datetime.date.today()` would be a lot more efficient

File: mail_cleanup/models/mail_cleanup.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+                failed += 1
+            count += 1
+        _logger.info("Marked %d email(s) as read on %s server %s; "
+                     "%d succeeded, %d failed.", count, server.type,
+                     server.name, (count - failed), failed)
+
+    def _purge_fetchmail_server(self, server, imap_server):
+        # Purging e-mails older than the purge date, if available
+        count, failed = 0, 0
+        purge_date = fields.Date.from_string(fields.Date.today())
```
Comment (lasley) [inline mail_cleanup/models/mail_cleanup.py]: `datetime.date.today()` would be a lot more efficient

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks for the submission @mdietrichc2c - comments inline

**Comment by mdietrichc2c** [comment]
@lasley changes done

**Review by lasley — APPROVED** [review]
LGTM thanks @mdietrichc2c

**Pull request merged**